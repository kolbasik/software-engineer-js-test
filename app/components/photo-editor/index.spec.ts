import {
	PhotoEditorState,
	photoEditorStateReducer,
	addProjects,
	deleteProject,
	updateProject,
	setActiveProject,
} from './index';

describe('PhotoEditor', () => {
	// Arrange
	const previousState: PhotoEditorState = {
		projects: [
			{
				id: 'TEST',
				canvas: {width: 5, height: 10},
				image: {src: 'none', width: 50, height: 100},
				matrix: {a: 1, b: 0, c: 0, d: 1, e: 0, f: 0},
			},
		],
		activeProjectId: undefined,
	};

	describe('addProjects', () => {
		it('should be handled', () => {
			// Arrange
			const project = {
				id: 'TEST2',
				canvas: {width: 5, height: 10},
				image: {src: 'none', width: 50, height: 100},
				matrix: {a: 1, b: 0, c: 0, d: 1, e: 0, f: 0},
			};

			// Act
			const nextState = photoEditorStateReducer(
				previousState,
				addProjects({
					projects: [project],
				}),
			);

			// Assert
			expect(nextState).not.toBe(previousState);
			expect(nextState.projects[1]).toBe(project);
			expect(nextState).toMatchSnapshot();
		});
	});

	describe('deleteProject', () => {
		it('should be handled', () => {
			// Arrange
			const project = previousState.projects[0];

			// Act
			const nextState = photoEditorStateReducer(
				previousState,
				deleteProject({
					projectId: project.id,
				}),
			);

			// Assert
			expect(nextState).not.toBe(previousState);
			expect(nextState.projects.length).toBe(0);
			expect(nextState).toMatchSnapshot();
		});
	});

	describe('updateProject', () => {
		it('should be handled', () => {
			// Arrange
			const project = {...previousState.projects[0]};

			// Act
			const nextState = photoEditorStateReducer(
				previousState,
				updateProject({
					project,
				}),
			);

			// Assert
			expect(nextState).not.toBe(previousState);
			expect(nextState.projects[0]).toBe(project);
			expect(nextState).toMatchSnapshot();
		});
	});

	describe('setActiveProject', () => {
		it('should be handled', () => {
			// Arrange
			const project = previousState.projects[0];

			// Act
			const nextState = photoEditorStateReducer(
				previousState,
				setActiveProject({
					activeProjectId: project.id,
				}),
			);

			// Assert
			expect(nextState).not.toBe(previousState);
			expect(nextState.activeProjectId).toBe(project.id);
			expect(nextState).toMatchSnapshot();
		});
	});
});
