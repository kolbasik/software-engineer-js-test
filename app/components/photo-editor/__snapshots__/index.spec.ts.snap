// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoEditor addProjects should be handled 1`] = `
Object {
  "activeProjectId": "TEST",
  "projects": Array [
    Object {
      "canvas": Object {
        "height": 10,
        "width": 5,
      },
      "id": "TEST",
      "image": Object {
        "height": 100,
        "src": "none",
        "width": 50,
      },
      "matrix": Object {
        "a": 1,
        "b": 0,
        "c": 0,
        "d": 1,
        "e": 0,
        "f": 0,
      },
    },
    Object {
      "canvas": Object {
        "height": 10,
        "width": 5,
      },
      "id": "TEST2",
      "image": Object {
        "height": 100,
        "src": "none",
        "width": 50,
      },
      "matrix": Object {
        "a": 1,
        "b": 0,
        "c": 0,
        "d": 1,
        "e": 0,
        "f": 0,
      },
    },
  ],
}
`;

exports[`PhotoEditor deleteProject should be handled 1`] = `
Object {
  "activeProjectId": undefined,
  "projects": Array [],
}
`;

exports[`PhotoEditor setActiveProject should be handled 1`] = `
Object {
  "activeProjectId": "TEST",
  "projects": Array [
    Object {
      "canvas": Object {
        "height": 10,
        "width": 5,
      },
      "id": "TEST",
      "image": Object {
        "height": 100,
        "src": "none",
        "width": 50,
      },
      "matrix": Object {
        "a": 1,
        "b": 0,
        "c": 0,
        "d": 1,
        "e": 0,
        "f": 0,
      },
    },
  ],
}
`;

exports[`PhotoEditor updateProject should be handled 1`] = `
Object {
  "activeProjectId": undefined,
  "projects": Array [
    Object {
      "canvas": Object {
        "height": 10,
        "width": 5,
      },
      "id": "TEST",
      "image": Object {
        "height": 100,
        "src": "none",
        "width": 50,
      },
      "matrix": Object {
        "a": 1,
        "b": 0,
        "c": 0,
        "d": 1,
        "e": 0,
        "f": 0,
      },
    },
  ],
}
`;
